pyramid_swagger/tween20.py

```diff
@@ -0,0 +1,216 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from collections import namedtuple
+import logging
+
+from pyramid.interfaces import IRoutesMapper
+
+from bravado_core.request import RequestLike, unmarshal_request
+from pyramid_swagger.exceptions import RequestValidationError
+from pyramid_swagger.exceptions import ResponseValidationError
+from pyramid_swagger.model import PathNotMatchedError
+from pyramid_swagger.tween import get_exclude_paths, should_exclude_request, \
+    _get_validation_context
+from pyramid_swagger.tween import validation_error
+
+
+log = logging.getLogger(__name__)
+
+
+class Settings(namedtuple(
+    'Settings',
+    [
+        'spec',
+        'validate_request',
+        'validate_response',
+        'validate_path',
+        'exclude_paths',
+        'exclude_routes',
+    ]
+)):
+
+    """A settings object for configuratble options.
+
+    :param spec: a :class:`bravado_core.spec.Spec`
+    :param validate_swagger_spec: check Swagger files for correctness.
+    :param validate_request: check requests against Swagger spec.
+    :param validate_response: check responses against Swagger spec.
+    :param validate_path: check if request path is in schema. If disabled
+        and path not found in schema, request / response validation is skipped.
+    :param exclude_paths: list of paths (in regex format) that should be
+        excluded from validation.
+    :rtype: namedtuple
+    :param exclude_routes: list of route names that should be excluded from
+        validation.
+    """
+
+
+def swagger_tween_factory(handler, registry):
+    """Pyramid tween for performing marshaling, transforming, and validating
+    Swagger 2.0 requests and responses.
+
+    :type handler: function
+    :type registry: :class:`pyramid.registry.Registry`
+    """
+    settings = load_settings(registry)
+    route_mapper = registry.queryUtility(IRoutesMapper)
+
+    def swagger_tween(request):
+        """
+        :type request: :class:`pyramid.request.Request`
+        """
+        route_info = route_mapper(request)
+
+        if should_exclude_request(settings, request, route_info):
+            return handler(request)
+
+        validation_context = _get_validation_context(registry)
+
+        try:
+            swaggerize_request(request, settings, route_info)
+        except (PathNotMatchedError, RequestValidationError) as exc:
+            if settings.validate_path:
```

Comment:
`RequestValidationError` generated should not be ignored if `validate_path` is set to `False`.